packages/browser/src/node/index.ts

```diff
@@ -18,6 +18,16 @@ export default (base = '/'): Plugin[] => {
     {
       enforce: 'pre',
       name: 'vitest:browser',
+      async resolveId(id, importer, options) {
+        if (id.includes('?'))
+          // Work-around for vite:resolve. The "import('./file.mjs?v=123')" fails to load "./file.mts" and throws error.
```

Comment:
I think this might introduce bugs with other plugins that rely on the query. We should only remove `?v=` here I think.

Reply:
Good catch. I'll change this to `.split('?v=')[0]` instead. There's still a small change that some library is using same query parameter and we end up removing that.

Reply:
Technically Vite also uses `v=` query for optimized dependencies. And this might actually be the reason it doesn't work 🤔 Did you try changing the query to something else? Like `vitest=` or something

Reply:
I'll change this to `?vitest=` so that we'll prevent conflicting with possible other queries. 

This got accidentally fixed in Vite but is still unreleased: https://github.com/vitejs/vite/pull/12571.

Reply:
Looks like this actually breaks watch mode. We'll need to use `?v=` query and pass that to Vite. Without the query parameter, or even with `?vitest=` query, the file updating triggers the re-load but file content is not updated. 

I'll start by adding a test case for watch mode so that this does not break again.